internal/bufferedwrites/upload_handler_test.go

```diff
@@ -340,3 +344,54 @@ func (t *UploadHandlerTest) TestCreateObjectChunkWriterIsCalledWithCorrectReques
 	err = t.uh.Upload(b)
 	require.NoError(t.T(), err)
 }
+
+func (t *UploadHandlerTest) TestDestroy() {
+	testCases := []struct {
+		name           string
+		uploadChClosed bool
+	}{
+		{
+			name:           "UploadChNotClosed",
+			uploadChClosed: false,
+		},
+		{
+			name:           "UploadChClosed",
+			uploadChClosed: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func() {
+			// Create some blocks.
+			var blocks []block.Block
```

Comment:
This is reused at multiple places. Shall we create a test helper to create 5 blocks?